src/main/java/io/appform/functionmetrics/FunctionTimerAspect.java

```diff
@@ -43,8 +52,35 @@ public Object around(ProceedingJoinPoint joinPoint) throws Throwable {
         final String methodName = Strings.isNullOrEmpty(monitoredFunction.method())
                                     ? callSignature.getName()
                                     : monitoredFunction.method();
-        log.trace("Called for class: {} method: {}", className, methodName);
-        final FunctionInvocation invocation = new FunctionInvocation(className, methodName);
+        final String[] parameterNames = methodSignature.getParameterNames();
+        String parameterString = "";
+        List<String> paramValues = Streams.zip(Arrays.stream(methodSignature.getMethod().getParameters()), Arrays.stream(joinPoint.getArgs()), Pair::new)
+                .filter(pair -> {
+                    MetricTerm metricTerm = pair.getKey().getAnnotation(MetricTerm.class);
+                    return metricTerm != null;
+                })
+                .map(pair -> {
+                    MetricTerm metricTerm = pair.getKey().getAnnotation(MetricTerm.class);
+                    return new Pair<>(metricTerm.order(), pair.getValue());
+                })
+                .sorted(Comparator.comparingInt(Pair::getKey))
+                .map(Pair::getValue)
+                .map(paramValue -> paramValue != null ? paramValue : "")
+                .map(paramValue -> paramValue instanceof String ? (String) paramValue : "")
```

Comment:
Merge the two above and reduce one extra function invocation.

Reply:
Move this up to before sorting. Map to a Pair<int, string>. Remove all useless crap and then get into sorting. Why waste cycles on sorting things that you might end up discarding later on.

Reply:
What if param is an enum? Sounds like the most useful case for me as the possible values are definitely bounded. Strings can still cause an explosion of metrices.

Reply:
Agreed that strings cause an explosion of metrics but there is nothing much that could be done. It is good to accept enums for parameter values. Let me see if I can do that.

Reply:
> Move this up to before sorting. Map to a Pair<int, string>. Remove all useless crap and then get into sorting. Why waste cycles on sorting things that you might end up discarding later on.

It makes the code ugly and a bit unreadable if I do the sorting later, I find this cleaner. Otherwise, I will have to deal with pairs everywhere and use setValue methods. I feel sorting ahead won't have much of an overhead since we are only sorting give or take less than 3-4 elements on average.

Reply:
For these kind of libraries, performance should be the main concern.